tests/monotouch-test/Foundation/NSOperatingSystemVersionTest.cs

```diff
@@ -0,0 +1,126 @@
+		public void NSVersionCompareObject ((Object obj, int result) sample)
+		{
+			NSOperatingSystemVersion v1 = new NSOperatingSystemVersion ((nint) 1, (nint) 2, (nint) 3);
+			Assert.AreEqual (v1.CompareTo (sample.obj), sample.result);
+		}
+
+
+		// Hash Tests
+
+		public static IEnumerable<NSOperatingSystemVersion> VersionCasesHashCode {
+			get {
+				yield return (new NSOperatingSystemVersion (majorVersion: (nint) 1, minorVersion: (nint) 2, patchVersionArg: (nint) 3));
+				yield return (new NSOperatingSystemVersion (majorVersion: (nint) 2));
+				yield return (new NSOperatingSystemVersion (majorVersion: (nint) 2, minorVersion: (nint) 5));
+				yield return (new NSOperatingSystemVersion (majorVersion: (nint) 2, minorVersion: (nint) 9, patchVersionArg: (nint) 10));
+			}
+		}
+
+		[Test, TestCaseSource("VersionCasesHashCode")]
+		public void NSVersionHashCode (NSOperatingSystemVersion version)
+		{
+			Assert.AreEqual (version.GetHashCode(), version.GetHashCode());
+		}
+
+		// ToString Tests
+
+		public static IEnumerable<(NSOperatingSystemVersion, string)> VersionCasesString {
+			get {
+				yield return (new NSOperatingSystemVersion (majorVersion: (nint) 1, minorVersion: (nint) 2, patchVersionArg: (nint) 3), "1.2.3");
+				yield return (new NSOperatingSystemVersion (majorVersion: (nint) 2), "2.0.0");
+				yield return (new NSOperatingSystemVersion (majorVersion: (nint) 2, minorVersion: (nint) 5), "2.5.0");
+				yield return (new NSOperatingSystemVersion (majorVersion: (nint) 2, minorVersion: (nint) 9, patchVersionArg: (nint) 10), "2.9.10");
+			}
+		}
+
+		[Test, TestCaseSource ("VersionCasesString")]
+		public void NSVersionToString ((NSOperatingSystemVersion version, string versionStr) sample)
+		{
+			Assert.AreEqual (sample.version.ToString(), sample.versionStr);
+		}
+
+		// ICompare Tests
+
+		public static IEnumerable<(NSOperatingSystemVersion, NSOperatingSystemVersion, int)> VersionCasesCompare {
+			get {
+				yield return (VersionOne: new NSOperatingSystemVersion (majorVersion: (nint) 1), VersionTwo: new NSOperatingSystemVersion(majorVersion: (nint)1), Result: 0);
+				yield return (VersionOne: new NSOperatingSystemVersion (majorVersion: (nint) 1), VersionTwo: new NSOperatingSystemVersion (majorVersion: (nint) 2), Result: -1);
+				yield return (VersionOne: new NSOperatingSystemVersion (majorVersion: (nint) 1, minorVersion: (nint) 1), VersionTwo: new NSOperatingSystemVersion (majorVersion: (nint) 1), Result: 1);
+				yield return (VersionOne: new NSOperatingSystemVersion (majorVersion: (nint) 1, minorVersion: (nint) 2, patchVersionArg: (nint) 2), VersionTwo: new NSOperatingSystemVersion (majorVersion: (nint) 1, minorVersion: (nint) 2, patchVersionArg: (nint) 3), Result: -1);
+			}
+		}
+
+		[Test, TestCaseSource("VersionCasesCompare")]
+		public void NSVersionCompare ((NSOperatingSystemVersion v1, NSOperatingSystemVersion v2, int res) sample)
+		{
+			Assert.AreEqual (sample.v1.CompareTo (sample.v2), sample.res);
+		}
+	}
+}
```

Comment:
```suggestion
}

```
That little red circle means that the file is missing a new line so don't forget to add that in for cs files